includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1129,7 +1129,9 @@ public function ajax_add_to_cart() {
 			define( 'WOOCOMMERCE_CART', true );
 		}
 
-		$subscription_types = [
+		$supported_product_types = [
+			'simple',
+			'variation',
 			'subscription',
 			'subscription_variation',
```

Comment:
I just wonder if we should get rid of this array altogether and replace it with `$this->supported_product_types()` which includes additional product types like `booking`, `bundle` and `composite`. 

However, I'm not sure whether these product types are supported by Google/Apple Pay or if they were excluded for a reason.

Reply:
I don't think we can use `$this->supported_product_types()` directly, in this scenario. I see that [in this code block](https://github.com/Automattic/woocommerce-payments/blob/fix/7695-total-0-for-variation-products-gpay/includes/class-wc-payments-payment-request-button-handler.php#L1149-L1156), we are handling the logic slightly differently than [in this other code block](https://github.com/Automattic/woocommerce-payments/blob/fix/7695-total-0-for-variation-products-gpay/includes/class-wc-payments-payment-request-button-handler.php#L1158-L1160).

If this array is renamed to `$supported_product_types` from `$subscription_types`, I think we should clean up a few things try and find a way to handle both scenarios.

I see that `$subscription_types` was added here: https://github.com/Automattic/woocommerce-payments/pull/5964 , with the intent of targeting "simpler" types (i.e.: products that don't have child variations).

I don't have a better naming for this array - maybe it could also be inlined, since it's not used anywhere else.